source/material_model/visco_plastic.cc

```diff
@@ -263,7 +263,12 @@ namespace aspect
               }
             }
 
-          // Step 1d: multiply the viscosity by a constant (default value is 1)
+          // Step 1d: compute viscosity from Peierls creep law and harmonically average with current viscosities
+          const double viscosity_peierls = peierls_creep.compute_viscosity(edot_ii, pressure, temperature_for_viscosity, j);
```

Comment:
It looks like you compute it even if it is not used, is that on purpose?

Reply:
Yep, this should go inside if (use_peierls_creep). I initialize viscosity_peierls as an NaN right before.